parse_test.go

```diff
@@ -1227,7 +1229,11 @@ func TestNewPrivateKey(t *testing.T) {
 		key.Algorithm = algo.name
 		privkey, err := key.Generate(algo.bits)
 		if err != nil {
-			t.Fatal(err)
+			if !algo.deprecated {
+				t.Fatal(err)
+			}
+			// deprecated algo, this must fail
+			continue
 		}
```

Comment:
If you want to keep testing these, there should be a separate block for `err == nil` like:

```go
} else {
  if algo.deprecated {
    t.Errorf(...)
  }
}
```

Reply:
i'll remove this; it's testing something else entirely (actually)